### DISCUSSION TITLE
Need to list all failed from certain  address.

### DISCUSSION THREAD

**Discussion by UEAB-BL** [postalserver/postal#3152] [Help with using Postal]
Currently sending emails from alot of different systems. I need a way to show all failed mails from a certain address or credential, so that the system administrators are able to view these and take some sort of action on it.

With our last mail server they recieved these as undelivered emails, but since postal doesnt work that way we need a way to work around it.

- I tried to filter, but cant figure out how to apply more than one filter, is it possible?
- I looked at the API but it doesnt seem able to list mails like that?
- Webhooks and make a seperate database from that? Seems really unneccesary.
- Look directly in the postal database maybe?

Any suggestions on what the best approach would be?

**Comment by willpower232** [comment]
either

> Webhooks and make a seperate database from that

or

> Look directly in the postal database maybe?

depending on how comfortable you are looking around in SQL databases.

Webhooks back to the originating system is the tidiest approach but that depends on the extensibility of the originating systems.

If you're already set up for undelivered emails, you could receive all webhooks to something you make which reformats the webhooks as undeliverable messages?

**Reply by UEAB-BL** [reply]
Thanks! I actually managed to get the filtering to work, not sure why it wasnt working before, but I restarted postal and then it worked.
Before the restart, I wasnt able to use more than one filter for some reason.

For any one with the same issues, I made a link like this for sys admins to use, and gave them logins to postal.
https://postal.mydomain/org/org/servers/postal-server/messages/outgoing?query=status%3A+hardfail+status%3Abounced+status%3Asoftfail+status%3Aheld+from%3Amyemail@domain.com

I might use webhooks for similar situations too, seems pretty clean and easy!

Thanks for the repsonse! =)
